File: DSA/LeetCode/TwoPointers/2486-Append-Characters-to-String-to-Make-Subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int sP = 0;
        int tP = 0;

        while (sP < s.length() && tP < t.length()) {
            if (s.charAt(sP) == t.charAt(tP)) {

                tP++;
            }
            sP++;

        }
        if (tP != t.length()) {
            return t.length() - tP;
        }
        return 0;
    }
}

File: DSA/LeetCode/TwoPointers/2486-Append-Characters-to-String-to-Make-Subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        sP = 0
        tP = 0

        while  sP < len(s) and tP <len(t):
            if s[sP] == t[tP]:
                tP +=1
            sP +=1
            
        if tP != len(t):
            return len(t) - tP
        
        return 0
        